client/src/Forms/DefineForm.js

```diff
@@ -1,13 +1,173 @@
 import React from 'react';
+import Form from './FormComponent';
+import {theme, ThemeProvider} from "../Libraries/ReactToolboxLibrary";
 
-import { add_definition_url } from "../urls";
+import '../stylesheets/defineForm.css';
 
-const DefineForm = () => {
+const snackbarMessage = "One or more required fields are blank";
+const submissionFailureMessage = "Oh no! Something went wrong, please try again.";
+
+const submissionSuccessMessage = (termName) => {
+  return `Your proposed definition for ${termName} was submitted successfully`
+};
+
+const defaultFieldValues = {
+  submitterName: "",
+  termName: "",
+  termDefinition: "",
+  termMeaningToSubmitter: "",
+  submitterIdentities: "",
+};
+
+export default class DefineForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      stateBar: false,
+      snackbarMessage: '',
+      ...defaultFieldValues
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(label, isRequired, showErrorInput, value) {
+    this.setState({[label]: value});
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    if(this.formChild.hasErrors()) {
+      this.setState({
+        stateBar: true,
+        snackbarMessage: snackbarMessage
+      });
+
+      return;
+    }
+
+    const state = this.state;
+
+    const payload = {
+      /* ???: submitterName */
```

Comment:
do we need to keep this comment?

Reply:
Good catch. I've double-checked the server code and it looks like we can map that to a `name` field.